orttraining/orttraining/python/ort_trainer.py

```diff
@@ -742,15 +765,38 @@ def train(self):
     def eval(self):
         self.is_train = False
 
+    def _update_onnx_model_initializers(self, state_tensors):
+        # replace the initializers with new value
+        new_weights = []
+        replace_indices = []
+        i = 0
+        for w in self.onnx_model_.graph.initializer:
+            if w.name in state_tensors:
+                new_weights.append(numpy_helper.from_array(state_tensors[w.name], w.name))
+                replace_indices.append(i)
+            i += 1
+        replace_indices.sort(reverse=True)
```

Comment:
Why is it necessary to delete the old initializers in reverse?

Reply:
if not, indices are shifted after each removal.